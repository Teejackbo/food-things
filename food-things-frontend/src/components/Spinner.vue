<template>
  <div v-if="fullScreen" class="wrapper">
    <transition name="fade">
      <div class="spinner"/>
    </transition>
  </div>
  <transition v-else name="fade">
    <div class="spinner"/>
  </transition>
</template>

<script>
export default {
  name: 'Spinner',
  props: {
    fullScreen: {
      type: Boolean,
      required: false,
    },
  },
}
</script>

<style lang="scss" scoped>
@import '../style/variables';

.wrapper {
  align-items: center;
  display: flex;
  height: 80vh;
  justify-content: center;
  width: 100%;
}

.spinner {
  animation: spin 1s linear infinite;
  border: 4px solid transparent;
  border-bottom-color: $primary;
  border-top-color: $primary;
  border-radius: 50%;
  height: 5em;
  width: 5em;
}

@keyframes spin {
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(360deg);
  }
}

.fade-enter-active,
.fade-leave-active {
  transition: opacity 0.5s;
}

.fade-enter,
.fade-leave-to {
  opacity: 0;
}
</style>
