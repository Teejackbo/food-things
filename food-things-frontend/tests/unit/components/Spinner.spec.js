import { shallowMount } from '@vue/test-utils'
import Spinner from '@/components/Spinner.vue'

describe('<Spinner />', () => {
  it('Should render a div with a class of spinner.', () => {
    const wrapper = shallowMount(Spinner)

    expect(wrapper.contains('div')).toBe(true)
    expect(wrapper.classes()).toContain('spinner')
    expect(wrapper.element).toMatchSnapshot()
  })

  it('Should render a div with a class of wrapper if passed :fullScreen="true".', () => {
    const wrapper = shallowMount(Spinner, {
      propsData: {
        fullScreen: true,
      },
    })

    expect(wrapper.contains('div')).toBe(true)
    expect(wrapper.classes()).toContain('wrapper')
    expect(wrapper.classes()).not.toContain('spinner')
    expect(wrapper.element).toMatchSnapshot()
  })

  it('Should render a div with a class of spinner inside the wrapper.', () => {
    const wrapper = shallowMount(Spinner, {
      propsData: {
        fullScreen: true,
      },
    })

    const spinner = wrapper.find('.spinner')
    expect(spinner).toBeDefined()
    expect(spinner.is('div')).toBe(true)
    expect(wrapper.element).toMatchSnapshot()
  })
})
