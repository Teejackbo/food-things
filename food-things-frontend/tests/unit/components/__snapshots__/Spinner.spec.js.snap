// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner Should render a div with a class of spinner inside the wrapper. 1`] = `
<div
  class="wrapper"
>
  <div
    class="spinner"
    name="fade"
  />
</div>
`;

exports[`Spinner Should render a div with a class of spinner. 1`] = `
<div
  class="spinner"
  name="fade"
/>
`;

exports[`Spinner Should render a div with a class of wrapper if passed :fullScreen="true". 1`] = `
<div
  class="wrapper"
>
  <div
    class="spinner"
    name="fade"
  />
</div>
`;
